File: pregunta 1/preg1.py
# -*- coding: utf-8 -*-
"""
Created on Mon Nov 29 21:31:34 2021

@author: Miguel
"""

def suma  (a,b):
    return a+b

def resta (a,b):
    return a-b
def multiplicacion (a,b):
    return a*b
def division (a,b):
    return a/b

op = 0
while (op != -1):
    print ("1.- Sumar\n2.- Restar\n3.- Multiplicar\n4.- Dividir>")
    op = int(input())
    print ("Numero 1:")
    x = int(input())
    print ("Numero 1:")
    y = int(input())
    if (op == 1 ):
        print ("resultado: ", suma(x, y))
    else: 
        if (op == 2):
            print ("resultado: ",resta(x, y))
        else: 
            if (op == 3):
                print ("resultado: ",multiplicacion(x, y))
            else: 
                if (op == 4):
                    print ("resultado: ",division(x, y))
                else : print("NA") 
            
           

File: pregunta 1/preg1.hs
getnum::IO Int
getnum = do
  s <- getLine
  return (read s)

suma :: Int -> Int -> Int
suma x y = x+y

resta :: Int -> Int ->Int
resta x y = x-y

multiplicacion :: Int -> Int ->Int
multiplicacion x y = x*y

division :: Int -> Int ->Int
division x y =  x `div` y 

main= do
putStr("1.- Sumar\n2.- Restar\n3.- Multiplicar\n4.- Dividir\n>\n")
opc <- getnum

putStr("Numero 1:\n")
x <- getnum
putStr("Numero 2:\n")
y <- getnum



let calculadora=if opc==1 then suma x y else if opc == 2 then resta x y else if opc == 3 then multiplicacion x y else division x y 

putStrLn("Resultado:"++show(calculadora)++"\n")

main
